Source/Core/UICommon/GameFile.cpp

```diff
@@ -387,6 +387,7 @@ std::string GameFile::GetExtension() const
 {
   std::string extension;
   SplitPath(m_file_path, nullptr, nullptr, &extension);
+  std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
```

Comment:
This matches the conversion code in Boot.cpp but this is actually dangerous – the behaviour of std::tolower is undefined if called with anything that cannot be represented using an unsigned char and isn't equal to EOF.

The standard way to deal with this pitfall is to convert to unsigned char *before* calling std::tolower(int): `[](unsigned char c) { return std::tolower(c); }`

Reply:
That seems like a rather awful footgun.  I think I'll instead create a `Common::ToLower` in `StringUtil.h` that uses `std::tolower(c, std::locale::classic())` and use that elsewhere.